source/features/raw-file-link.tsx

```diff
@@ -4,28 +4,26 @@ import delegate from 'delegate-it';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
+import GitHubURL from '../github-helpers/github-url';
 
 function handleMenuOpening(event: delegate.Event): void {
+	event.delegateTarget.classList.add('rgh-raw-file-link'); // Mark this as processed
 	const dropdown = event.delegateTarget.nextElementSibling!;
 
-	// Only if it's not already there
-	if (select.exists('.rgh-raw-file-link', dropdown)) {
-		return;
-	}
-
 	const viewFile = select<HTMLAnchorElement>('[data-ga-click^="View file"]', dropdown)!;
-	const url = viewFile.pathname.split('/');
-	url[3] = 'raw'; // Replaces 'blob'
+	const href = new GitHubURL(viewFile.href).assign(
+		{route: 'raw'}
+	).toString();
 
 	viewFile.after(
-		<a href={url.join('/')} className="pl-5 dropdown-item btn-link rgh-raw-file-link" role="menuitem">
+		<a href={href} className="pl-5 dropdown-item btn-link rgh-raw-file-link" role="menuitem">
 			View raw
 		</a>
 	);
 }
 
 function init(): void {
-	delegate(document, '.js-file-header-dropdown > summary', 'click', handleMenuOpening);
+	delegate(document, '.js-file-header-dropdown > summary:not(.rgh-raw-file-link)', 'click', handleMenuOpening);
```

Comment:
I’m not sure why this doesn’t listen to `toggle` instead

Reply:
Just tried it, seems not to work

Reply:
Summary doesn't `toggle`, you'll have to update the selector. Look at other features

Reply:
I can look but `enable-file-links-in-compare-view` uses the same thing

Reply:
Done should I do this to `enable-file-links-in-compare-view`

Reply:
Did it